src/main/java/org/folio/rest/impl/InvoiceHelper.java

```diff
@@ -147,15 +178,287 @@
         setSystemGeneratedData(invoiceFromStorage, invoice);
         return invoiceFromStorage;
       })
-      .thenCompose(invoiceFromStorage -> {
-        if (isTransitionToPaid(invoiceFromStorage, invoice)) {
-          return payInvoice(invoice);
-        } else {
-          return updateInvoiceRecord(invoice);
+      .thenCompose(invoiceFromStorage -> handleInvoiceStatusTransition(invoice, invoiceFromStorage))
+      .thenCompose(aVoid -> updateInvoiceRecord(invoice));
+  }
+
+  private void setSystemGeneratedData(Invoice invoiceFromStorage, Invoice invoice) {
+    invoice.withFolioInvoiceNo(invoiceFromStorage.getFolioInvoiceNo());
+  }
+
+  private CompletionStage<Void> handleInvoiceStatusTransition(Invoice invoice, Invoice invoiceFromStorage) {
+    if (isTransitionToApproved(invoiceFromStorage, invoice)) {
+      return approveInvoice(invoice);
+    } else if (isTransitionToPaid(invoiceFromStorage, invoice)) {
+      return payInvoice(invoice);
+    }
+    return VertxCompletableFuture.completedFuture(null);
+  }
+
+  private boolean isTransitionToApproved(Invoice invoiceFromStorage, Invoice invoice) {
+    return invoiceFromStorage.getStatus() == Invoice.Status.REVIEWED && invoice.getStatus() == Invoice.Status.APPROVED;
+  }
+
+  /**
+   * Handles transition of given invoice to Approved status.
```

Comment:
Maintainability: It's really helpful how the logic is detailed in the PR, if it is present here in the comments section, it would be helpful in the longer run

Reply:
Added doc comments